src/level_set/FESurfaceDistanceEvaluator.cpp

```diff
@@ -357,18 +358,297 @@ FESurfaceDistanceEvaluator::computeSignedDistance(int n_idx, int d_idx)
+                    // distance, then the cell is equidistant to multiple
+                    // elements, so add it to the set.
+                    if (MathUtilities<double>::equalEps(dist, min_dist))
+                    {
+                        vec_equidistant_pair.push_back(std::make_pair(proj, d_elem_face_normal[elem]));
+                    }
+                    else if (dist < min_dist)
+                    {
+                        // If a new minimal element is found, clear the
+                        // previous vector of pairs and simply keep this
+                        // one
+                        min_dist = dist;
+                        vec_equidistant_pair.clear();
+                        vec_equidistant_pair.push_back(std::make_pair(proj, d_elem_face_normal[elem]));
+                    }
+#endif
+
+#if (NDIM == 3)
+                    const auto proj_and_normal = getClosestPointandAngleWeightedNormal3D(P, elem);
+                    proj = proj_and_normal.first;
+                    dist = (P - proj).norm();
+
+                    // If the distance is the same as the minimal
+                    // distance, then the cell is equidistant to multiple
+                    // elements, so add it to the set.
+                    if (MathUtilities<double>::equalEps(dist, min_dist))
+                    {
+                        vec_equidistant_pair.push_back(std::make_pair(proj, proj_and_normal.second));
+                    }
+                    else if (dist < min_dist)
+                    {
+                        // If a new minimal element is found, clear the
+                        // previous vector of pairs and simply keep this
+                        // one
+                        min_dist = dist;
+                        vec_equidistant_pair.clear();
+                        vec_equidistant_pair.push_back(std::make_pair(proj, proj_and_normal.second));
+                    }
+#endif
+                }
+
+                // Determine the sign based on the angle-weighted pseudo normal vectors
+                // If the cell is equidistant to multiple elements, take
+                // the average pseudo normal from those elements
+                double sgn = 0.0;
+                const size_t vec_length = vec_equidistant_pair.size();
+                TBOX_ASSERT(vec_length > 0);
+                IBTK::VectorNd avg_unit_normal, avg_proj;
+                avg_unit_normal.setZero();
+                avg_proj.setZero();
+                for (const auto& elem_vec_pair : vec_equidistant_pair)
+                {
+                    const IBTK::VectorNd proj = elem_vec_pair.first;
+                    ;
+                    avg_proj += proj;
+
+                    const IBTK::VectorNd N = elem_vec_pair.second;
+                    avg_unit_normal += N;
+                }
+
+                // Take the average normal and normalize it
+                avg_unit_normal /= (double)vec_length;
```

Comment:
This should be 
` avg_unit_normal /= static_cast<double>(vec_length);`

I forgot to convert to C++ casting from C casting here.